frame/contracts/src/schedule.rs

```diff
@@ -0,0 +1,345 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate. If not, see <http://www.gnu.org/licenses/>.
+
+//! This module contains the cost schedule and supporting code that constructs a
+//! sane default schedule from a `WeightInfo` implementation.
+
+use crate::{Trait, WeightInfo};
+
+#[cfg(feature = "std")]
+use serde::{Serialize, Deserialize};
+use frame_support::weights::Weight;
+use sp_std::{marker::PhantomData, fmt};
+use codec::{Encode, Decode};
+
+/// How many API calls are executed in a single batch. The reason for increasing the amount
+/// of API calls in batches (per benchmark component increase) is so that the linear regression
+/// has an easier time determining the contribution of that component.
+pub const API_BENCHMARK_BATCH_SIZE: u32 = 100;
+
+/// Definition of the cost schedule and other parameterizations for wasm vm.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Clone, Encode, Decode, PartialEq, Eq)]
+pub struct Schedule<T: Trait> {
+	/// The type parameter is used in the default implementation.
+	pub phantom: PhantomData<T>,
+
+	/// Version of the schedule.
+	pub version: u32,
+
+	/// Weight of a growing memory by single page.
+	pub op_cost_grow_mem: Weight,
+
+	/// Weight of a regular operation.
+	pub op_cost_regular: Weight,
+
+	/// Weight of calling `seal_caller`.
+	pub api_cost_caller: Weight,
```

Comment:
Can you split all `api_cost_*` in separate struct and avoid prefix on all members here?

Reply:
Yeah I think this makes sense.

Reply:
Done.